api/v1beta1/conversion.go

```diff
@@ -18,12 +18,12 @@ package v1beta1
 
 import (
 	"context"
-	"fmt"
-	"github.com/apache/cloudstack-go/v2/cloudstack"
-	"github.com/pkg/errors"
+
 	corev1 "k8s.io/api/core/v1"
 	conv "k8s.io/apimachinery/pkg/conversion"
 	"sigs.k8s.io/cluster-api-provider-cloudstack/api/v1beta2"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/controllers"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/pkg/cloud"
```

Comment:
I'm nervous about adding these dependencies. It seems like we could easily introduce a dependency cycle, especially with the controller dependency. Could we instead define a new package like string_utils for defining the WithClusterSuffix method?

Otherwise LGTM